testutils/inject/digital_interrupt.go

```diff
@@ -36,12 +36,10 @@ func (d *DigitalInterrupt) ValueCap() []interface{} {
 	return d.valueCap
 }
 
-// Tick calls the injected Tick or the real version.
+// Tick calls the injected Tick.
 func (d *DigitalInterrupt) Tick(ctx context.Context, high bool, nanoseconds uint64) error {
 	d.tickCap = []interface{}{ctx, high, nanoseconds}
-	if d.TickFunc == nil {
-		return d.DigitalInterrupt.Tick(ctx, high, nanoseconds)
-	}
+
 	return d.TickFunc(ctx, high, nanoseconds)
```

Comment:
If `TickFunc` is nil, this is going to panic. ...but if you try using an injected interrupt in a test and don't set `TickFunc`, maybe your test _should_ panic, and this is okay. When someone encounters that situation, I wonder if it'll be obvious how to fix it.

(I don't have any actionable feedback here, just thinking about how people will react to this.)

Reply:
Yeah did think about this but i think its fine that it panics since you will need to implement the tick func.

Reply:
We can remove it in a test refactor later - I think the happy place is that you rely on an injected stream ticks function if you need to implement a digital interrupt.

You can also just return nil, or have the inject package helper look like the the Basic Interrupt helper.